### PR TITLE
[Spring] Add subcommand "az spring app connect".

### PR DESCRIPTION
**Pull request by LGDoor**
---

This checklist is used to make sure that common guidelines for a pull request are followed.

### General Guidelines

- [x] Have you run `azdev style <YOUR_EXT>` locally? (`pip install azdev` required)
- [x] Have you run `python scripts/ci/test_index.py -q` locally?

For new extensions:

- [ ] My extension description/summary conforms to the [Extension Summary Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/extensions/extension_summary_guidelines.md).

### About Extension Publish

There is a pipeline to automatically build, upload and publish extension wheels.  
Once your pull request is merged into main branch, a new pull request will be created to update `src/index.json` automatically.  
The precondition is to put your code inside this repository and upgrade the version in the pull request but do not modify `src/index.json`.

### CODE REVIEW

**Comment by yonzhan** [comment]
Spring

**Comment by zhoxing-ms** [comment]
Could you add some tests for those new commands?

**Comment by zhoxing-ms** [comment]
Please add the description of this PR in the `HISTORY.rst`. If you want to release a new extension version for this PR, please also upgrade the api-version defined in `setup.py`

File: src/spring/azext_spring/custom.py [hunk 1]
```diff
@@ -1462,3 +1467,49 @@ def app_insights_show(cmd, client, resource_group, name, no_wait=False):
...
+    profile = Profile(cli_ctx=cmd.cli_ctx)
+    creds, _, _ = profile.get_raw_token()
+    token = creds[1]
+
+    resource = client.services.get(resource_group, service)
+    hostname = resource.properties.fqdn
+    if not instance:
+        if not deployment.properties.instances:
+            raise CLIError("No instances found for deployment '{0}' in app '{1}'".format(
+                deployment.name, name))
```
Comment (zhoxing-ms) [inline src/spring/azext_spring/custom.py:1484]: Could you please use the specific error type from `azure.cli.core.azclierror` instead of `CLIError`?

Comment (LGDoor) [reply src/spring/azext_spring/custom.py]: Updated.

**Review by zhoxing-ms — APPROVED** [review]

**Review by zhoxing-ms — APPROVED** [review]

**Pull request merged**